src/sentry_plugins/bitbucket/endpoints/webhook.py

```diff
@@ -0,0 +1,181 @@
+from __future__ import absolute_import
+
+import dateutil.parser
+import hashlib
+import hmac
+import logging
+import six
+import re
+
+import ipaddress
+
+from django.db import IntegrityError, transaction
+from django.http import HttpResponse, Http404
+from django.utils.crypto import constant_time_compare
+from django.utils.decorators import method_decorator
+from django.views.decorators.csrf import csrf_exempt
+from django.views.generic import View
+from django.utils import timezone
+from simplejson import JSONDecodeError
+from sentry.models import (
+    Commit, CommitAuthor, Organization, Repository
+)
+from sentry.plugins.providers import RepositoryProvider
+from sentry.utils import json
+
+logger = logging.getLogger('sentry.webhooks')
+
+# Bitbucket Cloud IP range: https://confluence.atlassian.com/bitbucket/manage-webhooks-735643732.html#Managewebhooks-trigger_webhookTriggeringwebhooks
+BITBUCKET_IP_RANGE = ipaddress.ip_network(u'104.192.143.0/24')
+
+
+def is_anonymous_email(email):
+    # todo(maxbittker) investigate and improve behavior
+    return email[-25:] == '@users.noreply.bitbucket.com'
+
+
+def get_external_id(username):
+    # todo(maxbittker) investigate and improve behavior
+    return 'bitbucket:%s' % username
+
+
+class Webhook(object):
+    def __call__(self, organization, event):
+        raise NotImplementedError
+
+
+def parse_raw_user(raw):
+    # captures content between angle brackets
+    return re.search('(?<=<).*(?=>$)', raw).group(0)
+
+
+class PushEventWebhook(Webhook):
+    # https://confluence.atlassian.com/bitbucket/event-payloads-740262817.html#EventPayloads-Push
+    def __call__(self, organization, event):
+        authors = {}
+
+        # client = BitbucketClient()
+        try:
+            repo = Repository.objects.get(
+                organization_id=organization.id,
+                provider='bitbucket',
+                external_id=six.text_type(event['repository']['uuid']),
+            )
+        except Repository.DoesNotExist:
+            raise Http404()
+
+        for change in event['push']['changes']:
+            for commit in change['commits']:
+                if RepositoryProvider.should_ignore_commit(commit['message']):
```

Comment:
i realize this isnt changed behavior, so dont worry about addressing here, but IMO we should ALWAYS be storing commits. If our goal was to simply ignore them in release logs (or similar) we could do that elsewhere.